public/app/features/alerting/state/reducers.test.ts

```diff
@@ -283,13 +261,137 @@ describe('Notification channel', () => {
     },
   };
 
+  const notifiers: NotifierDTO[] = [
+    {
+      type: 'webhook',
+      name: 'webhook',
+      heading: 'Webhook settings',
+      description: 'Sends HTTP POST request to a URL',
+      info: '',
+      optionsTemplate: '',
+      options: [
+        {
+          element: 'input',
+          inputType: 'text',
+          label: 'Url',
+          description: '',
+          placeholder: '',
+          propertyName: 'url',
+          showWhen: { field: '', is: '' },
+          required: true,
+          validationRule: '',
+          secure: false,
+        },
+        {
+          element: 'select',
+          inputType: '',
+          label: 'Http Method',
+          description: '',
+          placeholder: '',
+          propertyName: 'httpMethod',
+          selectOptions: [
+            { value: 'POST', label: 'POST' },
+            { value: 'PUT', label: 'PUT' },
+          ],
+          showWhen: { field: '', is: '' },
+          required: false,
+          validationRule: '',
+          secure: false,
+        },
+        {
+          element: 'input',
+          inputType: 'text',
+          label: 'Username',
+          description: '',
+          placeholder: '',
+          propertyName: 'username',
+          showWhen: { field: '', is: '' },
+          required: false,
+          validationRule: '',
+          secure: false,
+        },
+        {
+          element: 'input',
+          inputType: 'password',
+          label: 'Password',
+          description: '',
+          placeholder: '',
+          propertyName: 'password',
+          showWhen: { field: '', is: '' },
+          required: false,
+          validationRule: '',
+          secure: true,
+        },
+      ],
+    },
+  ];
+
   describe('Load notification channel', () => {
-    it('should load basic case', () => {
+    it('should migrate non secure settings to secure fields', () => {
+      const payload = {
+        id: 2,
+        uid: '9L3FrrHGk',
+        name: 'Webhook test',
+        type: 'webhook',
+        isDefault: false,
+        sendReminder: false,
+        disableResolveMessage: false,
+        frequency: '',
+        created: '2020-08-28T08:49:24Z',
+        updated: '2020-08-28T08:49:24Z',
+        settings: {
+          autoResolve: true,
+          httpMethod: 'POST',
+          password: 'asdf',
+          severity: 'critical',
+          uploadImage: true,
+          url: 'http://localhost.webhook',
+          username: 'asdf',
+        },
+      };
+
+      reducerTester<NotificationChannelState>()
+        .givenReducer(notificationChannelReducer, { ...initialChannelState, notifiers: notifiers })
+        .whenActionIsDispatched(notificationChannelLoaded(payload))
+        .thenStateShouldEqual({
+          ...initialChannelState,
+          notifiers: notifiers,
+          notificationChannel: expected,
```

Comment:
Should verify expected has `secureSettings.password` set to "asdf" unless I misunderstand something?

Reply:
Yeah, but we need to set secureFields for the ui to render them as configured.

Reply:
we should clear the secureSettings.password right?

Reply:
> Yeah, but we need to set secureFields for the ui to render them as configured. This logic isn't on post but on get

Yes, on GET secureFields is returned and should be used to render them as configured. But was referring to when you load/get a legacy notification channel, then the old/legacy value should be displayed in input field in cleartext (if not password field). When saving this `settings.password` should be cleared and `secureSettings.password` populated. Then when fetching updated notification channel it will have secureFields set and UI will render configured as expected.

Loading notification channel:
1. If a secure input field `<property>` have a value in `settings.<property>`, put that value in `secureSettings.<property>` and set the `settings.<property> = null` .
2. If a secure input field `<property>` don't have a value in `settings.<property>` continue as is.

Does this make sense? :)

Reply:
@peterholmberg did you see my comment here ^?

Reply:
What data source is using secure properties? so we can test?